strawberryfields/io/xir_io.py

```diff
@@ -0,0 +1,294 @@
+            # module attribute, which contains the names
+            # of all defined quantum operations
+            if op.name in ops.__all__:
+                # get the quantum operation from the sf.ops module
+                gate = getattr(ops, op.name)
+            else:
+                raise NameError(f"Quantum operation {op.name!r} not defined!")
+
+            # create the list of regrefs
+            regrefs = [q[i] for i in op.wires]
+
+            if op.params:
+                # convert symbolic expressions to symbolic expressions containing the corresponding
+                # MeasuredParameter and FreeParameter instances.
+                if isinstance(op.params, dict):
+                    vals = sfpar.par_convert(op.params.values(), prog)
+                    params = dict(zip(op.params.keys(), vals))
+                    for key, val in params.items():
+                        if val[0] == "p" and val[1:].isdigit():
+                            params[key] = p[int(val[1:])]
+                    gate(**params) | regrefs  # pylint:disable=expression-not-assigned
+                else:
+                    params = []
+                    for param in op.params:
+                        if isinstance(param, Decimal):
+                            params.append(float(param))
+                        elif isinstance(param, (list, np.ndarray)):
+                            params.append(np.array(_listr(param)))
+                        elif isinstance(param, str) and param[0] == "p" and param[1:].isdigit():
+                            params.append(p[int(param[1:])])
+                        else:
+                            params.append(param)
+                    params = sfpar.par_convert(params, prog)
+                    gate(*params) | regrefs  # pylint:disable=expression-not-assigned
+            else:
+                if callable(gate):
+                    gate() | regrefs  # pylint:disable=expression-not-assigned,pointless-statement
+                else:
+                    gate | regrefs  # pylint:disable=expression-not-assigned,pointless-statement
+
+    prog._target = xir_prog.options.get("target", None)  # pylint: disable=protected-access
+
+    if "shots" in xir_prog.options:
+        prog.run_options["shots"] = xir_prog.options["shots"]
+    if "cutoff_dim" in xir_prog.options:
+        prog.backend_options["cutoff_dim"] = xir_prog.options["cutoff_dim"]
+
+    return prog
+
+
+def to_xir(prog, **kwargs):
+    """Convert a Strawberry Fields Program to an XIR Program.
+
+    Args:
+        prog (Program): the Strawberry Fields program
+
+    Keyword Args:
+        add_decl (bool): Whether gate and output declarations should be added to
+            the XIR program. Default is ``False``.
+        version (str): Version number for the program. Default is ``0.1.0``.
+
+    Returns:
+        xir.Program
+    """
+    version = kwargs.get("version", "0.1.0")
+    xir_prog = xir.Program(version=version)
```

Comment:
Why do we let users specify the XIR program version? Surely, this code will only support one version at a time (for the foreseeable future), right?
```suggestion
    xir_prog = xir.Program()
```

Reply:
Good point. I'm not sure why we'd want this, to be honest.